Makefile

```diff
@@ -210,32 +176,14 @@ compdb:
 tidy:
 	$(RUN) tidy
 
-.PHONY: clear_xcode_cache
-clear_xcode_cache:
-ifeq ($(BUILD), osx)
```

Comment:
Output can go to a variety of places depending on the build command, but as @1ec5 notes, none of those places are `~/Library/Developer/Xcode/DerivedData/{mapboxgl-app,osxapp}-*`. Sorting this out, and getting the output to _always_ go to `build`, is #4582. I hope to get to that as part of this PR.